data/shaders/opengl/planetrings.frag

```diff
@@ -16,10 +17,24 @@ void main(void)
 	vec4 col = vec4(0.0);
 	vec4 texCol = texture(texture0, texCoord0);
 
+	vec3 eyenorm = normalize(varyingEyepos.xyz);
+
 	for (int i=0; i<NUM_LIGHTS; ++i) {
 		float l = findSphereEyeRayEntryDistance(-vec3(texCoord1), vec3(uViewMatrixInverse * uLight[i].position), 1.0);
 		if (l <= 0.0) {
-			col = col + texCol*uLight[i].diffuse;
+			// first term: diffuse light phase (like in full/new moon)
+			float mu = dot(normalize(vec3(uLight[i].position)), eyenorm);
+			float diffuse = sqrt((1 - mu) / 2);
+
+			// second term: diffuse mie (scattering through ring)
+			float g = 0.76f;
+			float phaseThrough = 3.f / (8.f * 3.141592) * ((1.f - g * g) * (1.f + mu * mu)) / ((2.f + g * g) * pow(1.f + g * g - 2.f * g * mu, 1.5f));
+
+			// third term: reflect mie (imitate albedo >= 1.0)
+			float muRev = dot(-normalize(vec3(uLight[i].position)), eyenorm);
+			float phaseReflect = 3.f / (8.f * 3.141592) * ((1.f - g * g) * (1.f + muRev * muRev)) / ((2.f + g * g) * pow(1.f + g * g - 2.f * g * muRev, 1.5f));
```

Comment:
I'd prefer `phaseReflect` and `phaseThrough` equations be consolidated to a single function parameterized on `mu` / `muRev`, with a comment indicating the equation's source and ideally with a link to the article or resource the Mie phase equation is derived from.